### PR TITLE
Add support for complex gdal data type

### LINKED ISSUE #170: Can not load STK file
I downloaded `FLEVOL.STK` (https://earth.esa.int/documents/653194/658149/AIRSAR_Flevoland) file from https://earth.esa.int/web/polsarpro/data-sources/sample-datasets
Then used ArchGDAL to load it with:

```
using ArchGDAL
const AG = ArchGDAL
dataset = AG.read("FLEVOL.STK")
```

```
julia> dataset = AG.read("FLEVOL.STK")
GDAL Dataset (Driver: AirSAR/AirSAR Polarimetric Image)
File(s): 
  FLEVOL.STK

Dataset (width x height): 1024 x 750 (pixels)
Number of raster bands: 6
  Error showing value of type ArchGDAL.IDataset:
ERROR: KeyError: key GDAL.GDT_CFloat32 not found
Stacktrace:
 [1] getindex at ./dict.jl:467 [inlined]
 [2] ArchGDAL.IRasterBand(::Ptr{Nothing}; ownedby::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/types.jl:188
 [3] getband at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:606 [inlined]
 [4] show(::IOContext{REPL.Terminals.TTYTerminal}, ::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/base/display.jl:24
 [5] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::ArchGDAL.IDataset) at ./multimedia.jl:47
 [6] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:214
 [7] display(::REPL.REPLDisplay, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:218
 [8] display(::Any) at ./multimedia.jl:328
 [9] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [10] invokelatest at ./essentials.jl:709 [inlined]
 [11] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:238
 [12] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:223
 [13] (::REPL.var"#do_respond#54"{Bool,Bool,REPL.var"#64#73"{REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:822
 [14] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [15] invokelatest at ./essentials.jl:709 [inlined]
 [16] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2355
 [17] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1144
 [18] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:356
```

The error above seems related to display the dataset, so I add `;` then check the propertory of `dataset`
```
julia> using ArchGDAL

julia> const AG = ArchGDAL
ArchGDAL

julia> dataset = AG.read("FLEVOL.STK");

julia> AG.getdriver(dataset)
Driver: AirSAR/AirSAR Polarimetric Image

julia> AG.nraster(dataset)
6

julia> AG.width(dataset)
1024

julia> AG.height(dataset)
750

julia> gt = AG.getgeotransform(dataset)
ERROR: Failed to get geotransform
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/utils.jl:22 [inlined]
 [3] getgeotransform! at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:642 [inlined]
 [4] getgeotransform(::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:646
 [5] top-level scope at REPL[12]:1

```
The package did get some information of the file, but failed to load the other.

The python package of GDAL can load the file correctly:
```
In [1]: from osgeo import gdal

In [2]: dataset = gdal.Open("FLEVOL.STK")

In [3]: dataset.RasterCount
Out[3]: 6

In [4]: dataset.GetRasterBand(1)
Out[4]: <osgeo.gdal.Band; proxy of <Swig Object of type 'GDALRasterBandShadow *' at 0x0000021E23AC8FC0> >

In [5]: a = dataset.GetRasterBand(1)

In [6]: r = a.ReadAsArray()
```

The GDAL itself support this type of file (https://gdal.org/drivers/raster/airsar.html).
Is it a problem related to ArchGDAL.jl?

### PR DESCRIPTION
**Pull request by felixcremer**
This adds GDT_CFloat* to the supported types in the conversion and allows to load complex raster data. 
I am not sure, whether the test case in tests_rasterio.jl is at the correct place, but I wanted to test whether the reading and writing of complex data works because the test in test_types.jl also works currently without conversion.

This would partially close #170.

### CODE REVIEW

File: test/test_rasterio.jl [hunk 1]
```diff
@@ -275,3 +275,14 @@ const AG = ArchGDAL;
...
+@testset "Complex IO" begin 
+    a = rand(ComplexF32, 10,10)
+    outpath = "/vsimem/" * "testcomplex1.tif"
+    AG.create(outpath, driver=AG.getdriver("GTiff"), width=10, height=10, nbands=1,dtype=ComplexF32)  do ds
+        AG.write!(ds, a, 1)
+    end
+    AG.read(outpath) do ds_copy
+    @test AG.getband(ds_copy,1) == a
+    end
+end
```
Comment (yeesian) [inline test/test_rasterio.jl:288]: Can we add a new line here?

**Review by yeesian — APPROVED** [review]

File: test/test_rasterio.jl (hunk 1 above)
Comment (felixcremer) [reply test/test_rasterio.jl:291]: I just added, a few more lines in the testset, but I didn't see exactly, where you wanted them. 
I hope this is ok now.

Comment (visr) [reply test/test_rasterio.jl:291]: I believe he meant at the end of the file

Comment (felixcremer) [reply test/test_rasterio.jl]: Ahh yes that makes sense. 
I just added it now.

Comment (yeesian) [reply test/test_rasterio.jl]: Thanks!

**Comment by yeesian** [comment]
Thank you!

**Pull request merged**